core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java

```diff
@@ -525,6 +525,19 @@ public static String resolveHostName(String hostname) throws UnknownHostExceptio
     return InetAddress.getByName(hostname).getCanonicalHostName();
   }
 
+  /**
+   * Resolves a given hostname IP address.
+   *
+   * @param hostname the input hostname, which could be an alias
+   * @return the hostname IP address, or null if it is null or empty
+   * @throws UnknownHostException if the given hostname cannot be resolved
+   */
+  public static String resolveIpAddress(String hostname) throws UnknownHostException {
+    Preconditions.checkNotNull(hostname);
```

Comment:
as a best practice, we put the variable name as the second argument so that if the precondition fails, the error message will tell us the name of the null variable
```java
Preconditions.checkNotNull(hostname, "hostname");
```

Reply:
Done